staging/maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/workflow/AbstractStagingBuildActionMojo.java

```diff
@@ -47,41 +50,65 @@
   @Parameter(property = "stagingRepositoryId")
   private String stagingRepositoryId;
 
-  protected String[] getStagingRepositoryIds()
+  protected Map<String, String[]> getStagingRepositoryIds()
       throws MojoExecutionException
   {
-    String[] result = null;
+    Map<String, String[]> result = Maps.newHashMap();
     if (stagingRepositoryId != null) {
       // explicitly configured either via config or CLI, use that
-      result = Iterables.toArray(Splitter.on(",").split(stagingRepositoryId), String.class);
+      result.put(this.getNexusUrl(), Iterables.toArray(Splitter.on(",").split(stagingRepositoryId), String.class));
     }
-    if (result == null) {
+
+    if (result.isEmpty()) {
       // collect all the repositories we created
       final ArrayList<String> resultNames = new ArrayList<String>();
       final File stageRoot = getStagingDirectoryRoot();
-      final File[] localStageRepositories = stageRoot.listFiles();
-      if (localStageRepositories == null) {
+      final File [] nexusInstanceDirs = stageRoot.listFiles();
```

Comment:
Find the list of nexus url md5 folders, than walk through them for the profile ids one by one.